src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableEntityContext.cs

```diff
@@ -234,21 +252,21 @@ internal TState GetStateWithInjectedDependencies<TState>(Func<TState> constructo
                 throw new EntitySchedulerException($"Failed to construct entity state object: {e.Message}", e);
             }
 
-            if (this.State.EntityExists)
+            if (this.State.EntityExists
+                && this.CurrentStateAccess != StateAccess.Deleted)
             {
                 try
                 {
-                    JsonConvert.PopulateObject(this.State.EntityState, result);
+                    JsonConvert.PopulateObject(this.State.EntityState, result, this.dataConverter.MessageSettings);
```

Comment:
Hmm, not sure if this should be there. Are we meant to use MessageSettings for entity state serialization?